esp-wifi/src/ble/btdm.rs

```diff
@@ -178,38 +159,17 @@ unsafe extern "C" fn mutex_unlock(_mutex: *const ()) -> i32 {
 }
 
 unsafe extern "C" fn queue_create(len: u32, item_size: u32) -> *const () {
-    if len != 5 && item_size != 8 {
-        panic!("Unexpected queue spec {} {}", len, item_size);
-    }
-    &BT_INTERNAL_QUEUE as *const _ as *const ()
+    let ptr = crate::compat::common::create_queue(len as i32, item_size as i32);
+    ptr.cast()
 }
 
 unsafe extern "C" fn queue_delete(queue: *const ()) {
-    trace!("Unimplemented queue_delete {:?}", queue);
+    crate::compat::common::delete_queue(queue as *mut c_void)
```

Comment:
This is just a style nit, but I think we should avoid `*mut c_void` if a function is only used inside the library. If we used the queue's type here, it would make it very clear at the usage site that we are deleting a queue of the correct type or not.

Reply:
Ideally I'd prefer `&mut ConcurrentQueue` or whichever this one is :)

Reply:
But just `*mut QueueType` would work better IMO

Reply:
ah - we are talking about `queue` - not item 👍